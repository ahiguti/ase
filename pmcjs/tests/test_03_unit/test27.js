
use('Perl');
var callback = Perl.Eval(
  "sub { " +
  "  my $x = shift;" +
  "  $_ = 'underline';" + 
  "  $x->('arg1');" +
  "}");
function do_callback(n, f) {
  print(n);
  callback(f);
}

function foo_a(a) {
  print(a);
}
function foo_ul(_) {
  print(_);
}
function テスト_a(a) {
  print(a);
}
function テスト_ul(_) {
  print(_);
}

do_callback("anon_a", function(a) { print(a); });
do_callback("anon_ul", function(_) { print(_); });
do_callback("foo_a", foo_a);
do_callback("foo_ul", foo_ul);
do_callback("テスト_a", テスト_a);
do_callback("テスト_ul", テスト_ul);

