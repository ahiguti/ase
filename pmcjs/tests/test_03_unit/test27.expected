anon_a
arg1
anon_ul
underline
foo_a
arg1
foo_ul
underline
テスト_a
arg1
テスト_ul
underline
